js/src/BarsModel.ts

```diff
@@ -103,6 +103,10 @@ export class BarsModel extends markmodel.MarkModel {
                     if (isNaN(value)) {
                         value = 0;
                     }
+
+                    // so we don't pass NaNs to d3 scale
+                    y_elem[index] = isNaN(y_elem[index]) ? 0 : y_elem[index]
```

Comment:
Are there any side effects if the base of the bars isn't 0?

Reply:
Setting it to negative baseline did create an issue. Good catch! The weird thing is, I don't see the NaN issues anymore after removing this line. It looks like they were also caused by the transition naming difference.